File: Neanias.Accounting.Service.WebApp/src/app/core/query/service-sync.lookup.ts
import { IsActive } from '@app/core/enum/is-active.enum';
import { Lookup } from '@common/model/lookup';
import { Guid } from '@common/types/guid';

export class ServiceSyncLookup extends Lookup implements ServiceSyncFilter {
	ids: Guid[];
	excludedIds: Guid[];
	like: string;
	isActive: IsActive[];

	constructor() {
		super();
	}
}

export interface ServiceSyncFilter {
	ids: Guid[];
	excludedIds: Guid[];
	like: string;
	isActive: IsActive[];
}

File: Cite.Accounting.Service/Query/ServiceSyncLookup.cs
using Cite.Accounting.Service.Common;
using Cite.Tools.Data.Query;
using System;
using System.Collections.Generic;

namespace Cite.Accounting.Service.Query
{
	public class ServiceSyncLookup : Lookup
	{
		public List<Guid> Ids { get; set; }
		public String Like { get; set; }
		public List<IsActive> IsActive { get; set; }

		public ServiceSyncQuery Enrich(QueryFactory factory)
		{
			ServiceSyncQuery query = factory.Query<ServiceSyncQuery>();

			if (this.Ids != null) query.Ids(this.Ids);
			if (this.IsActive != null) query.IsActive(this.IsActive);
			if (!String.IsNullOrEmpty(this.Like)) query.Like(this.Like);

			this.EnrichCommon(query);

			return query;
		}
	}
}
